### DISCUSSION TITLE
Thank you

### DISCUSSION THREAD

**Discussion by fredericgboutin-yapla** [rectorphp/rector#8848] [General]
I'm just taking a moment to say thank you to everybody contributing to `rector`, but I suppose especially @TomasVotruba  

I've been using rector for more than a year now, migrating our code continually and still, this morning when reading the changelog for the latest release I noticed the existence of [`rector-phpunit`](https://github.com/rectorphp/rector-phpunit).

I bootstrapped it with the [Code Quality rule set](https://getrector.com/find-rule?activeRectorSetGroup=phpunit&rectorSet=phpunit-code-quality) and 💥 I discovered how bad `assertEquals` is compared to `assertSame` and how many false positives our unit-tests were reporting.

And this is just the latest story on a long list of nice things rector made us aware of.

So,

![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExNGlraGh2YzZ6OGJpeXR3Ym45ZWVmN2JuMzk0Zmx5dmp1aTZhb3hiMyZlcD12MV9naWZzX3NlYXJjaCZjdD1n/26gsjCZpPolPr3sBy/giphy.gif)

**Comment by TomasVotruba** [comment]
Thank you @fredericgboutin-yapla , this is amazing to read :heart: 

I'm glad you love the rules we made to make codebases more precise, reliable and joyful to work with... without devs really having to think about all those details :+1: 

We'll keep shipping more goodies :muscle:
